src/Scene/AnimationPlayer.js

```
@@ -0,0 +1,203 @@
+const FRAMERATE = 60;
+const INTERVAL_TIME = 1000 / FRAMERATE;
+// if is true console.log are enabled to sniff animation'state
+const debugAnimation = false;
+
+// player statut
+const PLAYER_STATE = {
+    // player is stopped
+    STOP: 0,
+    // player plays animation
+    PLAY: 1,
+    // player is at the end of an animation
+    END: 2,
+    // player is paused
+    PAUSE: 3,
+};
+
+const debugMsg =
+    [
+        'Stop',
+        'Play',
+        'End',
+        'Pause',
+    ];
+
+// if debugAnimation is true console.log are enabled to sniff animation'state
+var _DEBUG = null;
```

nits: let

---

And there are other occurrences of var below. It's not critical though, they behave nearly the same. But let has a Temporal Dead Zone, which is useful to fail fast.